### PR TITLE
fix(@vtmn/svelte): add unit param on `VtmnSkeleton`

### PR DESCRIPTION
**Pull request by Tlahey**
Linked to https://github.com/Decathlon/vitamin-web/issues/1249

Add a new input `unit` on `VtmnSkeleton`
Rules applied : 
- Unit = `%` by default
- Width = `100` by default
- If unit are not `['%', 'em', 'rem', 'vw', 'ch', 'px']` -> unit = `%` (default value)
- If width are `< 0` -> width = `100` (default value)

image

### CODE REVIEW

File: packages/sources/svelte/src/components/structure/VtmnSkeleton/test/vtmnSkeleton.spec.js [hunk 1]
```diff
@@ -43,4 +43,19 @@ describe('VtmnSkeleton', () => {
...
     expect(getSkeleton(container)).toHaveAttribute('data-nrt', 'test');
   });
+
+  test('Should change the unit', () => {
+    const { container } = render(VtmnSkeleton, { unit: 'px' });
+    expect(getSkeleton(container)).toHaveStyle('--skeleton-width:100px');
+  });
+
+  test('Should set the unit as % if unit not found', () => {
+    const { container } = render(VtmnSkeleton, { unit: 'toto' });
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/structure/VtmnSkeleton/test/vtmnSkeleton.spec.js:53]:
```suggestion
    const { container } = render(VtmnSkeleton, { unit: 'foo' });
```

File: packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte [hunk 2]
```diff
@@ -28,8 +38,14 @@
     className,
   );
 
+  $: computedWidth = width > 0 ? width : VTMN_DEFAULT_WIDTH;
```
[the remark this answers is not in the document]
Comment (Tlahey) [reply packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte:41]: Hello @lauthieb 
Yes, I hesitate between the 2 possibilities.

Why I choose width = 100, like browser, when you set an invalid value on width, it removed the width prop.
So, in fact, there are not set to 0 by default.

[the remark this answers is not in the document]
Comment (lauthieb) [reply packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte]: Hi @Tlahey, I understand your point.
As `width: 0px` is correct, I prefer:
`$: computedWidth = width >= 0 ? width : VTMN_DEFAULT_WIDTH;`
Can you please apply this change? Thanks.

File: packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte [hunk 3]
```diff
@@ -1,16 +1,26 @@
 <script>
   import { cn } from '../../../utils/classnames';
-  import { VTMN_SKELETON_SHAPE } from './enums';
+  import {
+    VTMN_SKELETON_SHAPE,
+    VTMN_SKELETON_UNIT,
+    VTMN_DEFAULT_DEFAULT__WIDTH,
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte:6]: To rename `VTMN_SKELETON_DEFAULT_WIDTH`
+ add `VTMN_SKELETON_DEFAULT_UNIT`

File: packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte [hunk 4]
```diff
@@ -28,8 +38,27 @@
...
 
+  let computedUnit;
+  let computedWidth;
+
+  $: {
+    if (
+      width < 0 ||
+      (unit && !Object.values(VTMN_SKELETON_UNIT).includes(unit))
+    ) {
+      computedWidth = 100;
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte:49]: Please use `VTMN_SKELETON_DEFAULT_WIDTH`

File: packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte [hunk 5]
```diff
@@ -28,8 +38,27 @@
...
+      computedUnit = VTMN_SKELETON_UNIT.PERCENT;
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/structure/VtmnSkeleton/VtmnSkeleton.svelte:50]: Please use `VTMN_SKELETON_DEFAULT_UNIT`

**Review by lauthieb — CHANGES_REQUESTED** [review]

**Review by lauthieb — DISMISSED** [review]
That's ok for me, thanks @Tlahey

**Review by lauthieb — APPROVED** [review]

**Review by thibault-mahe — APPROVED** [review]

**Pull request merged**